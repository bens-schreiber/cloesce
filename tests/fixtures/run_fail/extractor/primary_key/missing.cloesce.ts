// @ts-nocheck

@D1
class Foo {
  id: number; // missing @PrimaryKey
}
