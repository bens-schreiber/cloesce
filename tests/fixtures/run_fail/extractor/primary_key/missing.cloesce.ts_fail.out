==== CLOESCE ERROR ====
Error [MissingPrimaryKey]: Missing primary key on a model
Phase: TypeScript IDL Extraction
Context: Foo.
@D1
class Foo {
  id: number; // missing @PrimaryKey
}

Suggested fix: Add a primary key field to your model (e.g., `id: number`)
